sdk/cloudmachine/Azure.Projects.OpenAI/api/Azure.Projects.OpenAI.netstandard2.0.cs

```diff
@@ -17,6 +17,19 @@ public static void Add(this System.Collections.Generic.List<OpenAI.Chat.ChatMess
         public static OpenAI.Embeddings.EmbeddingClient GetOpenAIEmbeddingsClient(this System.ClientModel.Primitives.ConnectionProvider workspace, string? deploymentName = null) { throw null; }
         public static void Trim(this System.Collections.Generic.List<OpenAI.Chat.ChatMessage> messages) { }
     }
+    public partial class ChatProcessor
+    {
+        public ChatProcessor(OpenAI.Chat.ChatClient chat) { }
+        public ChatProcessor(OpenAI.Chat.ChatClient chat, OpenAI.Embeddings.EmbeddingClient? embeddings, Azure.Projects.OpenAI.ChatTools? tools = null) { }
+        public Azure.Projects.OpenAI.ChatTools? Tools { get { throw null; } set { } }
+        public Azure.Projects.OpenAI.EmbeddingsVectorbase? VectorDb { get { throw null; } set { } }
+        protected virtual OpenAI.Chat.ChatCompletion OnComplete(System.Collections.Generic.List<OpenAI.Chat.ChatMessage> conversation, string prompt) { throw null; }
+        protected virtual void OnGround(System.Collections.Generic.List<OpenAI.Chat.ChatMessage> conversation, string prompt) { }
+        protected virtual void OnLength(System.Collections.Generic.List<OpenAI.Chat.ChatMessage> conversation, OpenAI.Chat.ChatCompletion completion) { }
+        protected virtual void OnStop(System.Collections.Generic.List<OpenAI.Chat.ChatMessage> conversation, OpenAI.Chat.ChatCompletion completion) { }
+        protected virtual void OnToolCalls(System.Collections.Generic.List<OpenAI.Chat.ChatMessage> conversation, OpenAI.Chat.ChatCompletion completion) { }
```

Comment:
Why are these named like events?

Reply:
they are kind of events. and they follow naming convention for such methods. Events in fact are not supposed to be called with On

Reply:
The last 3 make sense since they are used in the context of handling the chat response. `OnComplete` and `OnGround` seem a bit more out of place to me.

Reply:
I see - I misunderstood the behavior, especially when creating a sub-type of ChatProcessor.